### PR TITLE
Add file validation rules

### PR DESCRIPTION
**Pull request by girardinsamuel**
- [x] Add negated message 
- [x] Add better parsing of input size (don't use bytes) => found a nice package to do that 💪 
```
>>> FileSize('1k')
... 1000
>>> FileSize('1K')
... 1024
>>> FileSize('1kib')
... 1024
>>> FileSize('1K', default_binary=False, case_sensitive=False)
... 1000
>>> FileSize('1 kibibyte')
... 1024
```
So we can for example use it like this : `file(["document"], size="15MB")` !

- [x] Find a way to define MIME types, currently the default module in Python 3 is using the official MIME types (as defined here https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types). But it might be difficult for user to define the rule with those mime types, it's easier for him to just give a list of extensions.

- [x] Finally I would like to add (as proposed) some meta rules (which are rules) but higher levels such as `image` or `video`.

```python
image(['avatar'], size='600KB')
video(['document'], size='15M')
```

### CODE REVIEW

**Comment by girardinsamuel** [comment]
@nioperas06 @josephmancuso 

Those new rules for files are ready for review 😄 👍 
Here is a quick doc (that I am gonna integrate into Masonite doc)
 
```python
"""
{
  'doc': my_file_path
}
"""
validate.file('doc') # check if valid file

validate.file('doc', 1024) # check valid file and max size is 1 Kilobyte (1024 bytes)
validate.file('doc', '1K') # check valid file and max size 1 Kilobyte (1024 bytes)
validate.file('doc', '1k') # check valid file and max size 1 Kilobyte (1024 bytes)
validate.file('doc', '1KB') # check valid file and max size 1 Kilobyte (1024 bytes)
validate.file('doc', '15M') # check valid file and max size 15 Megabytes

validate.file('doc', mimes=['jpg', 'png', 'gif', 'tiff']) # check valid file and of one of those types

validate.file('doc', size="4MB", mimes=['jpg', 'png']) # check valid file and max size 4 MB and of one of those types
```
And two shortcuts for now : 

```python
validate.image('doc') # check valid image

validate.video('doc') # check valid video

validate.image('doc', size="2MB") # check valid image and size !
```

File: src/masonite/validation/Validator.py [hunk 1]
```diff
@@ -791,6 +793,193 @@ def negated_message(self, attribute):
...
+
+
+class BaseFileValidation(BaseValidation):
+    def __init__(self, validations, messages={}, raises={}):
+        # internal state
+        self.file_check = True
+        self.size_check = True
+        self.mimes_check = True
+        self.all_clear = True
+        super().__init__(validations, messages=messages, raises=raises)
```
Comment (josephmancuso) [inline src/masonite/validation/Validator.py]: I usually always call the parent class first. this way the parent can't override the state. I will do this

File: src/masonite/validation/Validator.py [hunk 2]
```diff
@@ -791,6 +793,193 @@ def negated_message(self, attribute):
...
+
+    def passes(self, attribute, key, dictionary):
+        if not os.path.isfile(attribute):
+            self.file_check = False
+            # it does not make sense to continue checks if not a file
```
Comment (josephmancuso) [inline src/masonite/validation/Validator.py]: Code is self explanatory. No need for a comment like this

**Comment by josephmancuso** [comment]
@girardinsamuel added some outdated comments of some things I found as I fixed the code

**Comment by girardinsamuel** [comment]
> @girardinsamuel added some outdated comments of some things I found as I fixed the code

Thanks for the feedback! Totally agree !

**Pull request merged**